### ISSUE TITLE
CLI template, why do you open Base?

### ISSUE THREAD

**Issue by danielo515** [tmattio/spin#70]
Hello!
Thank you for this project that lets me easily get into native reason development.
Today I experienced some weird errors until I look quiet (I calm myself down and come back) and I realized the problem.
My problem was that I copy pasted some code from another project which was using `Sys.read_dir` and friends, and all those calls were erroring out.
I look for documentation, and Sys had those methods (dam, it compiled on the other project!) and they were not deprecated. I tried to search for my particular error, but nothing was found.

Then, I realize that my IDE was telling my that Sys, was on module Base, that gave me a good hint... I come to dune config and, yes, there was an open Base flag, this should be some kind of shadowing problem... 
Then I looked into Base.Sys docs and discovered that it was different from the Sys docs I was reading.
I don't know how, but I come to the conclusion that I needed to use Caml.Sys. Caml seems to be the root module, so in normal circumstances Sys should be directly accessible.

This is a very confusing situation for a newcomer and I'm very glad that I figured out, but can be very frustrating. So my question is, is opening Base by default that important? Wouldn't it be better to just open it on those places where it is required?

Thanks and regards

**Comment by citizen428** [comment]
As you may be aware there are several competing standard libraries for OCaml: Batteries, Base, Containers, to name a few. This is largely due to the fact that until relatively recently (maybe around 4.07) OCaml's standard library wasn't particularly complete. While that's slowly changing, many people are still using alternatives, and Base has been popularized by being used in [Real World OCaml](https://dev.realworldocaml.org/) (a decision quite a few people aren't too happy with). 

In short: many people probably expect *a* standard library replacement to be present and Base is a fairly popular choice.

**Comment by danielo515** [comment]
Thanks. I understand that it makes a lot of sense to include Base already on the dependencies, but knowing that it shadows some important core modules, why opening it as default? Why not explicitly open on the places where it is needed and anything else?

**Comment by tmattio** [comment]
Hi @danielo515, sorry that you had troubles setting up your project. Maybe we could add a line in the contributing guides of the templates saying that the project uses Base, with a link to the documentation.

As to why we open Base by default, this is the recommended way to use the library: https://github.com/janestreet/base#using-the-ocaml-standard-library-with-base
> The recommended way to build code using Base is as follows:
> ```ocaml
> $ ocamlc -open Base
> ```

Most people using Base will shadow the stdlib modules and use the `Caml` module, as you did, if they really need to access things that have been shadowed.

**Comment by danielo515** [comment]
Hello @tmattio 
Thanks for your comprehension. I really think that a line stating that the project uses base and a link will probably enough for most people. In my particular case I would have probably hit the same problem because I am kind of impatient for trying new things, but that is nothing everybody else should fix 🤣
Knowing that I'm referring to the CLI template, where folder and files operation is quite common an extra example (or command) that uses Caml.Sys.readdir or similar will also help.

Thank you all for your support and answers
